tests/testthat/test-generate_data.R

```diff
@@ -280,3 +280,38 @@ test_that("genFactor works.", {
   expect_length(genFactor(copy(dt), c("q1", "q2"), replace = TRUE), 4)
   expect_equal(genFactor(copy(dt), c("q1", "q2"), labels = labels), dt_res)
 })
+
+#genDummy ----
+test_that("genDummy throws errors.", {
+  oldSeed <- .Random.seed
+  set.seed(10076)
+  
+  d1 <- defData(varname = "rx", formula = "1;1", dist = "trtAssign")
+  d1 <- defData(d1, varname = "male", formula = .4, dist = "binary")
+  d1 <- defData(d1, varname = "z", formula = "0 - 1.2*rx - 1*male", dist = "nonrandom")
+  
+  dd <- genData(5, d1)
+  
+  # Initial data checks
+  expect_error(genDummy(varname = "rx", sep = ".", replace = FALSE))
+  expect_error(genDummy(dd, sep = ".", replace = FALSE))
+  
+  # Check if data table exists
+  expect_error(genDummy(dt, varname = "rx", sep = ".", replace = FALSE))
```

Comment:
Just a warning - "dt" is actually a function name in base R (relating to the student-t density). It is possible to create a variable with that name - and I used to do that. But, it isn't great practice. In any case, probably best not to use that anywhere except when referring to the function itself. "DT" is a fine alternative, or anything else that is not a function.